src/lib/dnssd/Discovery_ImplPlatform.cpp

```diff
@@ -631,7 +631,8 @@ CHIP_ERROR DiscoveryImplPlatform::ResolveNodeId(const PeerId & peerId)
 {
     // Resolve requests can only be issued once DNSSD is initialized and there is
     // no caching currently
-    VerifyOrReturnError(mState == State::kInitialized, CHIP_ERROR_INCORRECT_STATE);
+    // When the DNSSD is kInitializing, can also try to resolve.
+    VerifyOrReturnError(mState != State::kUninitialized, CHIP_ERROR_INCORRECT_STATE);
```

Comment:
Could you point to some documentation stating this is ok? are all platforms ok to issue requests before async initialization completed?

Reply:
I am not sure that all platforms can try resolve. All platforms will still return errors when they cannot be resolved. Platform-related issues should not be handled here.